system/java_certs.py

```diff
@@ -0,0 +1,171 @@
+
+    # Use remote certificate from remote host and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, data=fetch_out, check_rc=False)
+    if not rc:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        if 'already exists' in import_out:
+            return module.exit_json(changed=False, msg=import_out,
+                rc=0, cmd=import_cmd, stdout_lines=import_out)
+        else:
+            return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+
+def import_cert_path(module, path, keystore_path, keystore_pass):
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -file '%s'" % (keystore_path, keystore_pass, path)
+
+    # Use local certificate from local path and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, check_rc=False)
+    if not rc:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        if 'already exists' in import_out:
+            return module.exit_json(changed=False, msg=import_out,
+                rc=0, cmd=import_cmd, stdout_lines=import_out)
+        else:
+            return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+
+def delete_cert(module, keystore_path, keystore_pass, alias):
+    del_cmd = "keytool -delete -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Delete SSL certificate from keystore
+    (rc, del_out, del_err) = module.run_command(del_cmd, check_rc=True)
+
+    return module.exit_json(changed=True, msg=del_out,
+        rc=rc, cmd=del_cmd, stdout_lines=del_out)
+
+def main():
+    argument_spec = dict()
+    argument_spec.update(dict(
+            cert_url = dict(required=False),
+            cert_path = dict(required=False),
+            cert_alias = dict(required=False),
+            cert_port = dict(required=False, default='443'),
+            keystore_path = dict(required=False),
+            keystore_pass = dict(required=False, default='changeit'),
+            state = dict(required=False, default='present', choices=['present', 'absent'])
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive = [
+                                ['cert_url', 'cert_path']
+                             ],
+    )
+
+    filter = {}
+    url = module.params.get('cert_url')
+    path = module.params.get('cert_path')
+    port = module.params.get('cert_port')
+    cert_alias = module.params.get('cert_alias') or url
+
+    keystore_path = module.params.get('keystore_path')
+    keystore_pass = module.params.get('keystore_pass')
+    state = module.params.get('state')
+
+    if state == 'absent':
+        delete_cert(module, keystore_path, keystore_pass, cert_alias)
```

Comment:
You should probably check whether the cert exists in the keystore before trying to remove it - if it doesn't exist, state=present is true and you can do nothing (and return changed=False)